### PR TITLE
expression: make some unstable test serial

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### What is changed and how it works?

Some Operations will change the global variables. If parallel, it will lead to many problems. 

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: expression/integration_serial_test.go [hunk 1]
```diff
@@ -0,0 +1,1493 @@
...
+	"github.com/pingcap/tidb/util/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestIssue17727(t *testing.T) {
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+
+	tk := testkit.NewTestKit(t, store)
+	orgEnable := plannercore.PreparedPlanCacheEnabled()
```
Comment (hawkingrei) [inline expression/integration_serial_test.go:38]: For example here

File: expression/integration_serial_test.go [hunk 2]
```diff
@@ -0,0 +1,1493 @@
...
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(id int, value set ('a','b','c') charset utf8mb4 collate utf8mb4_bin default 'a,b ');")
+	tk.MustExec("drop table t")
+	tk.MustExec("create table test(id int, value set ('a','b','c') charset utf8mb4 collate utf8mb4_general_ci default 'a,B ,C');")
+}
+
+func TestIssue20268(t *testing.T) {
+	collate.SetNewCollationEnabledForTest(true)
+	defer collate.SetNewCollationEnabledForTest(false)
```
Comment (hawkingrei) [inline expression/integration_serial_test.go:90]: For example, here

**Comment by hawkingrei** [comment]
@bb7133 @tangenta @tisonkun PTAL

**Review by AilinKid — APPROVED** [review]
LGTM

**Review by tisonkun — COMMENTED** [review]
Although the change looks good, don't these tests already run in serial?

```go
var _ = SerialSuites(&testIntegrationSerialSuite{})
```

**Comment by hawkingrei** [comment]
> Although the change looks good, don't these tests already run in serial?
> 
> ```go
> var _ = SerialSuites(&testIntegrationSerialSuite{})
> ```

but like 

```
var _ = Suite(&testIntegrationSuite{})

func (s *testIntegrationSuite) TestIssue17727(c *C) {
```

it is not in serial. But it still changes a global variable.

**Review by tisonkun — APPROVED** [review]

**Pull request merged**